internal/repository/index.go

```diff
@@ -476,18 +476,30 @@ func (idx *Index) merge(idx2 *Index) error {
 	}
 
 	packlen := len(idx.packs)
+	// first append packs as they might be accessed when looking for duplicates below
+	idx.packs = append(idx.packs, idx2.packs...)
+
 	// copy all index entries of idx2 to idx
 	for typ := range idx2.byType {
 		m2 := &idx2.byType[typ]
 		m := &idx.byType[typ]
-		m2.foreach(func(entry *indexEntry) bool {
-			// packIndex is changed as idx2.pack is appended to idx.pack, see below
-			m.add(entry.id, entry.packIndex+packlen, entry.offset, entry.length)
+		m2.foreach(func(e2 *indexEntry) bool {
+			found := false
```

Comment:
Please add a comment the the `merge` method which mentions that duplicate entries are removed.

The next few lines also needs a comment. And I'd prefer these to be move into a method like `hasPackedBlob` or similar. At first glance it's not obvious at all what the foreach inside another foreach does.

Reply:
is changed.